modules/text2vec-cohere/clients/cohere.go

```diff
@@ -172,6 +179,48 @@ func (v *vectorizer) getCohereUrl(ctx context.Context, baseURL string) string {
 	return v.urlBuilder.url(passedBaseURL)
 }
 
+func (v *vectorizer) GetApiKeyHash(ctx context.Context, config moduletools.ClassConfig) [32]byte {
+	key, err := v.getApiKey(ctx)
+	if err != nil {
+		return [32]byte{}
+	}
+	return sha256.Sum256([]byte(key))
+}
+
+func (v *vectorizer) GetVectorizerRateLimit(ctx context.Context) *modulecomponents.RateLimits {
+	rpm := DefaultRPM
+
+	if rpmS := v.getValueFromContext(ctx, "X-Cohere-Ratelimit-RequestPM-Embedding"); rpmS != "" {
```

Comment:
this won't work, headers need to be in a Word format: `X-Cohere-Ratelimit-Requestpm-Embedding`. This is how go stores headers in context. You can pass them in request like `X-Cohere-RateLimit-RequestPM-Embedding` but it gets parsed by go and placed in context in this format: `X-Cohere-Ratelimit-Requestpm-Embedding`

Reply:
It definitely works, checked with the debugger and there are unit tests. But I can rename it if you prefer it - however I find it more readable this way

Reply:
hmm, are you sure? I remember that we had bugs in modules where I had to change the capitalization of this header name explicitly to this `Word` notation. I will check this, maybe with `go1.21` it works differently.

Reply:
I just started the server and put a debugger breakpoint in this code. It read the variable succesfully

Reply:
But if you think it is risky I will rename

Reply:
indeed it works now...